src/pages/DetailsPage/DetailsPage.js

```diff
@@ -3,43 +3,41 @@ import FavorizeButton from '../../components/FavorizeButton/FavorizeButton'
 import BackButton from '../../components/BackButton/BackButton'
 import { PageContainer } from '../../components/PageContainer/PageContainer'
 import { useParams } from 'react-router'
-import { useRecoilState, useResetRecoilState } from 'recoil'
+import { useRecoilState } from 'recoil'
 import { detailedPodcastState } from '../../states'
 import podcasts from '../../test_data_space.json'
-import { useEffect } from 'react'
+import { useEffect, useState } from 'react'
+import LoadingScreen from '../../components/Loading Screen/LoadingScreen'
 
 export default function DetailsPage() {
   const { id } = useParams()
   const [podcastDetails, setPodcastDetails] = useRecoilState(
     detailedPodcastState
   )
-  const reset = useResetRecoilState(detailedPodcastState)
+  const [isLoading, setIsLoading] = useState(true)
 
   useEffect(() => {
     setPodcastDetails(podcasts.find(podcast => podcast.id === Number(id)))
-
-    return () => {
-      reset()
-    }
+    setTimeout(() => setIsLoading(false), 400)
     // eslint-disable-next-line react-hooks/exhaustive-deps
   }, [id])
```

Comment:
If the `id` changes, `loading` is not reset. 
Consider reset `loading` state in this side effect.

Reply:
Very good point. I added the reset.